### PR TITLE
feat: initial GitHub Action

### LINKED ISSUE #183: GIthub Action for Bomctl
Users need to be able to quickly leverage bomctl in their CICD workflows

### PR DESCRIPTION
**Pull request by jhoward-lm**
## Description

This PR is for the initial implementation of the bomctl GitHub Action.

Closes bomctl/bomctl#183

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] workflow (test-action.yml)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/scripts/install.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+run_install
+
+if [[ ! -x $install_path ]]; then
+  exit_with_error "bomctl executable not found at ${install_path} or not executable"
+fi
+
+log_info "Successfully installed bomctl to\n\t${install_path}"
+
+# DEBUG
+log_info "readlink -f ${INSTALL_DIR} ==> $(readlink -f "${INSTALL_DIR}")"
```
Comment (jhoward-lm) [inline .github/scripts/install.sh]: Remove debug

**Review by ashearin — APPROVED** [review]

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  steps:
+    - name: Install bomctl
+      id: install
+      shell: bash
+      env:
+        INSTALL_DIR: ${{ inputs.install-dir }}
+        BOMCTL_VERSION: ${{ inputs.bomctl-version }}
+      run: .github/scripts/install.sh
+
+    - name: Run bomctl command
```
Comment (idunbarh) [inline action.yml:97]: Should this be optional if a `command` is provided? I assume there will be folks who want to install `bomctl` but then specify their own workflow script commands.

Comment (jhoward-lm) [reply action.yml:97]: It defaults to just running `bomctl version` if a command isn't specified, pretty standard default behavior and a quick check to make sure the correct binary was installed. Does that answer the question?

Comment (idunbarh) [reply action.yml:97]: Yep! Thanks for the explanation.

**Review by idunbarh — APPROVED** [review]

**Pull request merged**